web/src/toga_web/widgets/dateinput.py

```diff
@@ -0,0 +1,65 @@
+import datetime
+
+from toga_web.libs import create_proxy
+
+from .base import Widget
+
+
+def py_date(native_value):
+    return datetime.date.fromisoformat(native_value)
+
+
+def native_date(py_date_obj):
+    return py_date_obj.isoformat()
+
+
+class DateInput(Widget):
+    def create(self):
+        self.native = self._create_native_widget("sl-input")
+        self.native.type = "date"
+        self.native.value = native_date(datetime.date.today())
+        self.native.addEventListener("sl-blur", create_proxy(self.dom_onblur))
+
+    def dom_onblur(self, event):
```

Comment:
Minor consistency thing - the event is `sl-blur`, so the event handler should be `dom_sl_blur`